### PR TITLE
EMER [ENG-850] Trustedform Refactor

### PR DESCRIPTION
**Pull request by heathdutton**

### CODE REVIEW

**Review by cykonetic — CHANGES_REQUESTED** [review]
I believe getEnhancerFieldArray is an abstract method that needs to be defined, even if it returns an empty array

**Review by cykonetic — APPROVED** [review]
Looks good. I see where the methods moved to.

File: Integration/TrustedFormIntegration.php [hunk 1]
```diff
@@ -185,4 +193,50 @@ protected function getFingers(Lead $lead)
 
         return $fingers;
     }
+
+    /**
+     * Get the type of authentication required for this API.  Values can be none, key, oauth2 or callback
+     * (will call $this->authenticationTypeCallback).
```
Comment (scottshipman) [inline Integration/TrustedFormIntegration.php:199]: update docbloc to say it returns basic?

**Review by scottshipman — COMMENTED** [review]
see comment about docbloc.

**Review by scottshipman — APPROVED** [review]

**Pull request merged**